src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -121,6 +121,10 @@ private function compileConstFetch(Node\Expr\ConstFetch $constNode)
      */
     private function compileClassConstFetch(Node\Expr\ClassConstFetch $node, CompilerContext $context)
     {
+        if ($node->class instanceof Node\Expr || $node->name instanceof Node\Expr\Error) {
```

Comment:
This would require also a test to reach this code location. This may as well be an assumption that is hard to code within the types we have available.

Reply:
`$node->class` cannot ever be a `Node\Expr` here, but `Node\Expr\ClassConstFetch::$class` can be. I suppose the error node may be possible, if you allow errors in `PhpParser`.

Reply:
I'll change to a `@var` docblock